codec-http/src/test/java/io/netty/handler/codec/http/HttpHeadersTest.java

```diff
@@ -76,8 +76,8 @@ public void testAddSelf() {
         headers.add(headers);
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testSetSelf() {
+    @Test
```

Comment:
I would prefer we keep the current semantic. If we want to change it we should discuss and ensure it is the same in 4.0, 4.1 and master